application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -355,4 +357,36 @@ public String getTagName() {
             return tagName;
         }
     }
+
+    /**
+     * Retrieves a {@link HelpThreadsRecord} object from the database based on the specified channel
+     * ID.
+     * <p/>
+     * This method queries for a record in the HELP_THREADS table where the CHANNEL_ID column
+     * matches the provided channelID param. It returns an Optional containing the matching record.
+     *
+     * @param channelID is channel unique identifier
+     *
+     * @return An Optional containing the HelpThreadsRecord for the specified channel ID, or an
+     *         empty Optional if no matching record is found.
+     *
+     * @throws DataAccessException if something is wrong with database
+     *
+     */
+    Optional<HelpThreadsRecord> getThreadRecordByChannelId(final long channelID) {
+
+        logger.debug("Looking for thread-record using channel ID: {}", channelID);
+
+        try {
+
+            return database.read(ctx -> ctx.selectFrom(HELP_THREADS)
+                .where(HELP_THREADS.CHANNEL_ID.eq(channelID))
+                .limit(1)
+                .fetchOptional());
+
+        } catch (DataAccessException ex) {
+            logger.warn("Something went wrong while trying to fetch data from thread-record table");
+            return Optional.empty();
+        }
```

Comment:
optional should not be used to indicate error.
let the exception naturally happen, dont catch it.

also, ur javadoc says that the exception happens (for an outside user) in case of issue - but thats not true right now, as u caught and swallowed it. just remove the try-catch-log thing entirely and then adjust ur doc